advertools/spider.py

```diff
@@ -949,6 +949,14 @@ def crawl(url_list, output_file, follow_links=False,
             else:
                 setting = '='.join([key, str(val)])
             settings_list.extend(['-s', setting])
+    meta_list = []
```

Comment:
Why is `meta` a list? Isn't it a dictionary?

Reply:
Looking at the `scrapy runspider` command. It seems that it takes parameters with the -s Global options. 

Basically, the below code block is handling "meta" (no matter if the data type is a dict, list, set, tuple) and putting into a list "meta_list" for `scrapy runspider
` Command. 
```
meta_list = []
    if meta is not None:
        for key, val in meta.items():
            if isinstance(val, (dict, list, set, tuple)):
                meta = '='.join([key, json.dumps(val)])
            else:
                meta = '='.join([key, str(val)])
            meta_list.extend(['-s', meta])

```

Example:
Running below: 
```
url_list = [
    "https://example.com/",
]
output_file = "/home/odoo/main-dev/odoon/scraping/advertools_contrib/advertools/code_recipes/example_crawl_1.jl
**meta = {"playwright": True,"playwright_include_page": True}**


for url in url_list:
    adv.crawl(
        url,
        output_file,
        follow_links=True,
        custom_settings={
            "DOWNLOAD_HANDLERS": {
                "http": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
                "https": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
            },
            "TWISTED_REACTOR": "twisted.internet.asyncioreactor.AsyncioSelectorReactor",
            "ROBOTSTXT_OBEY": False,
            "CLOSESPIDER_TIMEOUT": 10,
            "CONCURRENT_REQUESTS_PER_DOMAIN": 8,
            "CLOSESPIDER_PAGECOUNT": 4,
            "DEPTH_LIMIT": 1,
            "USER_AGENT": "Mozilla/5.0 (iPad; CPU OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148",
        },
        meta= meta
    )
```

Generates below command:

`['scrapy', 'runspider', '/home/odoo/main-dev/odoon/scraping/advertools_contrib/advertools/spider.py', '-a', 'url_list=https://example.com/', '-a', 'allowed_domains=example.com', '-a', 'follow_links=True', '-a', 'exclude_url_params=None', '-a', 'include_url_params=None', '-a', 'exclude_url_regex=None', '-a', 'include_url_regex=None', '-a', 'css_selectors=None', '-a', 'xpath_selectors=None', '-o', '/home/odoo/main-dev/odoon/scraping/advertools_contrib/advertools/code_recipes/example_crawl_1.jl', '-s', 'DOWNLOAD_HANDLERS={"http": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler", "https": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler"}', '-s', 'TWISTED_REACTOR=twisted.internet.asyncioreactor.AsyncioSelectorReactor', '-s', 'ROBOTSTXT_OBEY=False', '-s', 'CLOSESPIDER_TIMEOUT=10', '-s', 'CONCURRENT_REQUESTS_PER_DOMAIN=8', '-s', 'CLOSESPIDER_PAGECOUNT=4', '-s', 'DEPTH_LIMIT=1', '-s', 'USER_AGENT=Mozilla/5.0 (iPad; CPU OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148', **'-s', 'playwright=True', '-s', 'playwright_include_page=True**']
`
BTW, it is the same approach as the existing "custom_settings" implementation.

Reply:
Below is the log showing scrapy-playwright.

2022-08-25 10:09:45 [scrapy-playwright] INFO: Starting download handler

![image](https://user-images.githubusercontent.com/5569632/186643376-1f22b1f2-93c7-428f-99ac-79d3d99e1ee6.png)

Reply:
The meta parameter is not included in the generated command.
It is converted to custom settings, which it is not. Not all parameters are custom settings, some have the `-a` flag, and the meta parameter is not a custom setting.

This command converted the meta into custom settings, defeating the whole purpose of the meta implementation.

Reply:
> The meta parameter is not included in the generated command. It is converted to custom settings, which it is not. Not all parameters are custom settings, some have the `-a` flag, and the meta parameter is not a custom setting.
> 
> This command converted the meta into custom settings, defeating the whole purpose of the meta implementation.

Let me further look into it. 

But I have tested it multiple times like adding proxy into the meta and the request was successful as I compared source IPs at the destination server.

Reply:
> '-s', 'playwright=True', '-s', 'playwright_include_page=True'

This part that is showing in the call to `crawl` is basically like running this code:
```python
adv.crawl(DOMAIN, OUTPUTFILE, custom_settings={'playwright': True, 'playwright_include_page': True})
```

In this case, if this is working properly, then we don't need the `meta` parameter.

Did you check if the behavior works correctly with the desired outcome from using Playwright?